packages/core/test/lib/console-child.js

```diff
@@ -0,0 +1,86 @@
+const assert = require("chai").assert;
+const fs = require("fs-extra");
+const path = require("path");
+const tmp = require("tmp");
+const TruffleConfig = require("@truffle/config");
+const { deriveConfigEnvironment } = require("../../lib/command-utils");
+
+let config;
+
+function createSandbox(source) {
+  if (!fs.existsSync(source)) {
+    throw new Error(`Sandbox failed: source: ${source} does not exist`);
+  }
+
+  const tempDir = tmp.dirSync({ unsafeCleanup: true });
+  fs.copySync(source, tempDir.name);
+  const config = TruffleConfig.load(
+    path.join(tempDir.name, "truffle-config.js"),
+    {}
+  );
+  return config;
+}
+
+describe("console-child", function () {
+  before("Create a sandbox", function () {
+    config = createSandbox(
+      path.join(__dirname, "..", "sources", "console-child")
+    );
+  });
+
+  describe("deriveConfigEnvironment", function () {
+    it("network has provider", function () {
+      config.network = "crazyTimeNetwork";
```

Comment:
In all these cases I would prefer that you not mutate `config` and instead use string literals as arguments as I've mentioned in a couple of places.